tools/swift-format/Sources/Rules/VarDeclSyntax+Convenience.swift

```diff
@@ -0,0 +1,37 @@
+import SwiftSyntax
+
+extension VariableDeclSyntax {
+
+  /// Returns array of all identifiers listed in the declaration.
+  var identifiers: [IdentifierPatternSyntax] {
+    var ids: [IdentifierPatternSyntax] = []
+    for binding in bindings {
+      guard let id = binding.pattern as? IdentifierPatternSyntax else { continue }
+      ids.append(id)
+    }
+    return ids
+  }
+
+  /// Returns the first identifier.
+  var identifier: IdentifierPatternSyntax {
+    return identifiers[0]
+  }
+
+  /// Returns the type of the variable, if explicitly stated in the declaration.
+  var type: TypeSyntax? {
```

Comment:
There's one thing that worries me about the implementation of this property and `initializer` below. Because of the loop, it looks like you're returning the first type/initializer that _you see_, not the first one _period_. What would happen if I had a variable decl written like this?

```swift
let x: String?, y = 5
```

I think `type` would return `String?` here and `initializer` would return `5`, which isn't what we want.

It would probably be better to name these `firstType` and `firstInitializer`, for the reasons I mentioned above for `identifier`, so that they always return the type/initializer associated with the first binding, even if it's nil.

Reply:
Oh right, I keep only thinking in terms of a single variable declaration. Thanks!